src/jllvm/vm/Interpreter.cpp

```diff
@@ -17,47 +17,165 @@
 
 #include "VirtualMachine.hpp"
 
-std::uint64_t jllvm::Interpreter::executeMethod(const Method& method, std::uint16_t& offset,
-                                                InterpreterContext& /*context*/)
+namespace
 {
-    Code* code = method.getMethodInfo().getAttributes().find<Code>();
-    llvm::ArrayRef<char> codeArray = code->getCode();
-    auto curr = ByteCodeIterator(codeArray.drop_front(offset).data(), offset);
+/// Tag returned when interpreting an instruction to jump to a new bytecode offset.
+struct SetPC
+{
+    std::uint16_t newPC;
+};
 
-    /// Tag returned when interpreting an instruction to jump to a new bytecode offset.
-    struct SetPC
-    {
-        std::uint16_t newPC;
-    };
+/// Tag returned when interpreting an instruction to continue to the next instruction in the bytecode.
+struct NextPC
+{
+};
 
-    /// Tag returned when interpreting an instruction to continue to the next instruction in the bytecode.
-    struct NextPC
-    {
-    };
+/// Tag returned when interpreting an instruction to stop interpretation and return a result.
+struct ReturnValue
+{
+    std::uint64_t value;
 
-    /// Tag returned when interpreting an instruction to stop interpretation and return a result.
-    struct ReturnValue
+    template <jllvm::InterpreterValue T>
+    ReturnValue(T value) : value(llvm::bit_cast<jllvm::NextSizedUInt<T>>(value))
     {
-        std::uint64_t value;
-    };
+    }
+};
+} // namespace
+
+void jllvm::Interpreter::escapeToJIT()
+{
+    m_virtualMachine.unwindJavaStack(
+        [&](JavaFrame frame) { m_virtualMachine.getJIT().doI2JOnStackReplacement(frame, *frame.getByteCodeOffset()); });
+    llvm_unreachable("not possible");
+}
+
+std::uint64_t jllvm::Interpreter::executeMethod(const Method& method, std::uint16_t& offset,
+                                                InterpreterContext& context)
+{
+    const ClassFile& classFile = *method.getClassObject()->getClassFile();
+    Code* code = method.getMethodInfo().getAttributes().find<Code>();
```

Comment:
Since `find` can technically return a `nullptr` an assert would be fitting here.